python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -351,6 +362,26 @@ def _apply_default_remote_args(ray_remote_args: Dict[str, Any]) -> Dict[str, Any
     def get_autoscaling_actor_pools(self) -> List[AutoscalingActorPool]:
         return [self._actor_pool]
 
+    def _manage_actor_restarting_state(self, actor):
+        actor_state = actor._get_local_state()
+        if actor_state is None:
+            # actor._get_local_state can return None if the state is Unknown
+            return
+        elif actor_state != gcs_pb2.ActorTableData.ActorState.ALIVE:
+            # If an actor is not ALIVE, it's a candidate to be marked as a
+            # restarting actor.
+            assert actor_state is gcs_pb2.ActorTableData.ActorState.RESTARTING
```

Comment:
This is not future-proof as adding new actor state will break this assumption

Reply:
Yeah, adding more states to actor will break this, but asserting here catches adding new states. Not sure, if there is a better hook for this.

Reply:
If this is only supposed to be handling cases of RESTARTING actors make this branch only accept hese